modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/nextcloud_strategies.rb

```diff
@@ -37,10 +37,24 @@ module NextcloudStrategies
             ::Storages::Peripherals::StorageInteraction::AuthenticationStrategies::BasicAuth.strategy
           end
 
-          UserBound = ->(user:) do
-            ::Storages::Peripherals::StorageInteraction::AuthenticationStrategies::OAuthUserToken
-              .strategy
-              .with_user(user)
+          UserBound = ->(user:, storage:) do
```

Comment:
Would if be worth making `UserBound` into a full blown object? 🤔

Reply:
How would it look like? Do you mind writing down in pseudo code some of the interface you have in mind?

Reply:
Not very different than it looks like now, but since it is getting smarter and smarter, the fact that it is a lambda kinda bothers me. We can even keep the signature.

```
Very::Good::Name::UserBound.call(user:, storage:)
```